### PR TITLE
[Refactor] refactor Coordinator and CoordinatorPrepare

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Refactor:
1. Refactor the `Coordinator` into two parts: `prepare` and `exec`

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 645 / 989 (65.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPrepare.java|644|988|65.18%|[161, 165, 169, 173, 181, 185, 189, 193, 197, 201, 209, 213, 221, 225, 233, 237, 241, 245, 249, 253, 257, 261, 273, 277, 281, 289, 293, 332, 333, 334, 339, 340, 341, 342, 343, 344, 351, 352, 353, 354, 355, 356, 357, 359, 360, 361, 362, 363, 369, 370, 371, 409, 410, 411, 412, 413, 414, 421, 426, 427, 428, 476, 477, 478, 479, 481, 482, 483, 484, 486, 517, 518, 520, 521, 522, 523, 525, 567, 599, 600, 601, 602, 603, 604, 612, 613, 614, 616, 617, 618, 619, 620, 626, 628, 629, 631, 633, 634, 635, 637, 638, 640, 699, 711, 720, 721, 736, 740, 744, 753, 893, 894, 896, 909, 917, 918, 919, 920, 922, 924, 926, 927, 933, 934, 935, 955, 956, 960, 983, 1014, 1015, 1067, 1068, 1070, 1072, 1073, 1074, 1076, 1077, 1079, 1082, 1083, 1085, 1088, 1089, 1091, 1092, 1093, 1094, 1096, 1097, 1100, 1101, 1102, 1104, 1105, 1106, 1107, 1108, 1109, 1110, 1111, 1115, 1116, 1117, 1118, 1119, 1120, 1122, 1123, 1124, 1125, 1126, 1127, 1131, 1167, 1168, 1169, 1180, 1181, 1182, 1186, 1192, 1193, 1194, 1195, 1196, 1197, 1198, 1199, 1200, 1202, 1203, 1204, 1205, 1206, 1207, 1208, 1209, 1214, 1215, 1217, 1221, 1222, 1224, 1231, 1237, 1238, 1244, 1245, 1256, 1257, 1302, 1372, 1409, 1418, 1435, 1477, 1478, 1479, 1490, 1491, 1492, 1493, 1494, 1495, 1497, 1498, 1499, 1501, 1502, 1512, 1562, 1563, 1565, 1566, 1567, 1568, 1569, 1571, 1572, 1573, 1575, 1577, 1579, 1581, 1584, 1585, 1587, 1634, 1635, 1636, 1637, 1638, 1639, 1640, 1642, 1643, 1645, 1646, 1647, 1648, 1649, 1651, 1652, 1653, 1654, 1655, 1656, 1658, 1659, 1660, 1664, 1665, 1666, 1668, 1669, 1670, 1672, 1673, 1674, 1675, 1676, 1677, 1678, 1679, 1680, 1682, 1683, 1684, 1685, 1686, 1688, 1689, 1690, 1691, 1692, 1694, 1695, 1696, 1733, 1734, 1756, 1757, 1758, 1759, 1760, 1764, 1765, 1766, 1767, 1768, 1770, 1771, 1772, 1773, 1774, 1784, 1785, 1786, 1787, 1788, 1789, 1791, 1792, 1793, 1794, 1795, 1797, 1852, 1854, 1855, 1856, 1857, 1858, 1859, 1860, 1862, 1863, 1865, 1866, 1867, 1912, 1913]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 645 / 990 (65.15%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPrepare.java|644|989|65.12%|[161, 165, 169, 173, 181, 185, 189, 193, 197, 201, 209, 213, 221, 225, 233, 237, 241, 245, 249, 253, 257, 261, 273, 277, 281, 289, 293, 332, 333, 334, 339, 340, 341, 342, 343, 344, 351, 352, 353, 354, 355, 356, 357, 359, 360, 361, 362, 363, 369, 370, 371, 409, 410, 411, 412, 413, 414, 421, 426, 427, 428, 476, 477, 478, 479, 481, 482, 483, 484, 486, 517, 518, 520, 521, 522, 523, 525, 567, 599, 600, 601, 602, 603, 604, 612, 613, 614, 616, 617, 618, 619, 620, 626, 628, 629, 631, 633, 634, 635, 637, 638, 640, 699, 711, 720, 721, 736, 740, 744, 753, 893, 894, 896, 909, 917, 918, 919, 920, 922, 924, 926, 927, 933, 934, 935, 955, 956, 960, 983, 1014, 1015, 1067, 1068, 1070, 1072, 1073, 1074, 1076, 1077, 1079, 1082, 1083, 1085, 1088, 1089, 1091, 1092, 1093, 1094, 1096, 1097, 1100, 1101, 1102, 1104, 1105, 1106, 1107, 1108, 1109, 1110, 1111, 1115, 1116, 1117, 1118, 1119, 1120, 1122, 1123, 1124, 1125, 1126, 1127, 1131, 1167, 1168, 1169, 1180, 1181, 1182, 1186, 1192, 1193, 1194, 1195, 1196, 1197, 1198, 1199, 1200, 1202, 1203, 1204, 1205, 1206, 1207, 1208, 1209, 1214, 1215, 1217, 1221, 1222, 1224, 1231, 1237, 1238, 1244, 1245, 1256, 1257, 1302, 1372, 1409, 1418, 1435, 1477, 1478, 1479, 1490, 1491, 1492, 1493, 1494, 1495, 1497, 1498, 1499, 1501, 1502, 1512, 1562, 1563, 1565, 1566, 1567, 1568, 1569, 1571, 1572, 1573, 1575, 1577, 1579, 1581, 1584, 1585, 1587, 1634, 1635, 1636, 1637, 1638, 1639, 1640, 1642, 1643, 1645, 1646, 1647, 1648, 1649, 1651, 1652, 1653, 1654, 1655, 1656, 1658, 1659, 1660, 1664, 1665, 1666, 1668, 1669, 1670, 1672, 1673, 1674, 1675, 1676, 1677, 1678, 1679, 1680, 1682, 1683, 1684, 1685, 1686, 1688, 1689, 1690, 1691, 1692, 1694, 1695, 1696, 1726, 1735, 1736, 1758, 1759, 1760, 1761, 1762, 1766, 1767, 1768, 1769, 1770, 1772, 1773, 1774, 1775, 1776, 1786, 1787, 1788, 1789, 1790, 1791, 1793, 1794, 1795, 1796, 1797, 1799, 1854, 1856, 1857, 1858, 1859, 1860, 1861, 1862, 1864, 1865, 1867, 1868, 1869, 1914, 1915]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by Seaven — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 645 / 990 (65.15%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPrepare.java|644|989|65.12%|[161, 165, 169, 173, 181, 185, 189, 193, 197, 201, 209, 213, 221, 225, 233, 237, 241, 245, 249, 253, 257, 261, 273, 277, 281, 289, 293, 323, 324, 325, 330, 331, 332, 333, 334, 335, 342, 343, 344, 345, 346, 347, 348, 350, 351, 352, 353, 354, 360, 361, 362, 399, 400, 401, 402, 403, 404, 411, 416, 417, 418, 466, 467, 468, 469, 471, 472, 473, 474, 476, 507, 508, 510, 511, 512, 513, 515, 557, 589, 590, 591, 592, 593, 594, 602, 603, 604, 606, 607, 608, 609, 610, 616, 618, 619, 621, 623, 624, 625, 627, 628, 630, 689, 701, 710, 711, 726, 730, 734, 743, 883, 884, 886, 899, 907, 908, 909, 910, 912, 914, 916, 917, 923, 924, 925, 945, 946, 950, 973, 1004, 1005, 1057, 1058, 1060, 1062, 1063, 1064, 1066, 1067, 1069, 1072, 1073, 1075, 1078, 1079, 1081, 1082, 1083, 1084, 1086, 1087, 1090, 1091, 1092, 1094, 1095, 1096, 1097, 1098, 1099, 1100, 1101, 1105, 1106, 1107, 1108, 1109, 1110, 1112, 1113, 1114, 1115, 1116, 1117, 1121, 1157, 1158, 1159, 1170, 1171, 1172, 1176, 1182, 1183, 1184, 1185, 1186, 1187, 1188, 1189, 1190, 1192, 1193, 1194, 1195, 1196, 1197, 1198, 1199, 1204, 1205, 1207, 1211, 1212, 1214, 1221, 1227, 1228, 1234, 1235, 1246, 1247, 1292, 1364, 1401, 1410, 1427, 1469, 1470, 1471, 1482, 1483, 1484, 1485, 1486, 1487, 1489, 1490, 1491, 1493, 1494, 1504, 1554, 1555, 1557, 1558, 1559, 1560, 1561, 1563, 1564, 1565, 1567, 1569, 1571, 1573, 1576, 1577, 1579, 1626, 1627, 1628, 1629, 1630, 1631, 1632, 1634, 1635, 1637, 1638, 1639, 1640, 1641, 1643, 1644, 1645, 1646, 1647, 1648, 1650, 1651, 1652, 1656, 1657, 1658, 1660, 1661, 1662, 1664, 1665, 1666, 1667, 1668, 1669, 1670, 1671, 1672, 1674, 1675, 1676, 1677, 1678, 1680, 1681, 1682, 1683, 1684, 1686, 1687, 1688, 1718, 1727, 1728, 1750, 1751, 1752, 1753, 1754, 1758, 1759, 1760, 1761, 1762, 1764, 1765, 1766, 1767, 1768, 1778, 1779, 1780, 1781, 1782, 1783, 1785, 1786, 1787, 1788, 1789, 1791, 1846, 1848, 1849, 1850, 1851, 1852, 1853, 1854, 1856, 1857, 1859, 1860, 1861, 1906, 1907]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 642 / 986 (65.11%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPrepare.java|641|985|65.08%|[159, 163, 167, 171, 179, 183, 187, 191, 195, 203, 207, 215, 219, 227, 231, 235, 239, 243, 247, 251, 255, 267, 271, 275, 283, 287, 317, 318, 319, 324, 325, 326, 327, 328, 329, 336, 337, 338, 339, 340, 341, 342, 344, 345, 346, 347, 348, 354, 355, 356, 393, 394, 395, 396, 397, 398, 405, 410, 411, 412, 460, 461, 462, 463, 465, 466, 467, 468, 470, 501, 502, 504, 505, 506, 507, 509, 551, 583, 584, 585, 586, 587, 588, 596, 597, 598, 600, 601, 602, 603, 604, 610, 612, 613, 615, 617, 618, 619, 621, 622, 624, 683, 695, 704, 705, 720, 724, 728, 737, 877, 878, 880, 893, 901, 902, 903, 904, 906, 908, 910, 911, 917, 918, 919, 939, 940, 944, 967, 998, 999, 1051, 1052, 1054, 1056, 1057, 1058, 1060, 1061, 1063, 1066, 1067, 1069, 1072, 1073, 1075, 1076, 1077, 1078, 1080, 1081, 1084, 1085, 1086, 1088, 1089, 1090, 1091, 1092, 1093, 1094, 1095, 1099, 1100, 1101, 1102, 1103, 1104, 1106, 1107, 1108, 1109, 1110, 1111, 1115, 1151, 1152, 1153, 1164, 1165, 1166, 1170, 1176, 1177, 1178, 1179, 1180, 1181, 1182, 1183, 1184, 1186, 1187, 1188, 1189, 1190, 1191, 1192, 1193, 1198, 1199, 1201, 1205, 1206, 1208, 1215, 1221, 1222, 1228, 1229, 1240, 1241, 1286, 1358, 1395, 1404, 1421, 1463, 1464, 1465, 1476, 1477, 1478, 1479, 1480, 1481, 1483, 1484, 1485, 1487, 1488, 1498, 1548, 1549, 1551, 1552, 1553, 1554, 1555, 1557, 1558, 1559, 1561, 1563, 1565, 1567, 1570, 1571, 1573, 1620, 1621, 1622, 1623, 1624, 1625, 1626, 1628, 1629, 1631, 1632, 1633, 1634, 1635, 1637, 1638, 1639, 1640, 1641, 1642, 1644, 1645, 1646, 1650, 1651, 1652, 1654, 1655, 1656, 1658, 1659, 1660, 1661, 1662, 1663, 1664, 1665, 1666, 1668, 1669, 1670, 1671, 1672, 1674, 1675, 1676, 1677, 1678, 1680, 1681, 1682, 1712, 1721, 1722, 1744, 1745, 1746, 1747, 1748, 1752, 1753, 1754, 1755, 1756, 1758, 1759, 1760, 1761, 1762, 1772, 1773, 1774, 1775, 1776, 1777, 1779, 1780, 1781, 1782, 1783, 1785, 1840, 1842, 1843, 1844, 1845, 1846, 1847, 1848, 1850, 1851, 1853, 1854, 1855, 1900, 1901]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|[]|

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPrepare.java [hunk 1]
```diff
@@ -0,0 +1,1917 @@
...
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+public class CoordinatorPrepare {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPrepare.java:82]: CoordinatorPrepare is not a good name. Prepare is a verb. use Preprocessor instead?

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 642 / 993 (64.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPrepare.java|641|992|64.62%|[154, 158, 162, 170, 174, 178, 182, 190, 194, 202, 206, 214, 218, 222, 226, 230, 234, 238, 242, 250, 254, 258, 262, 270, 274, 305, 306, 307, 312, 313, 314, 315, 316, 317, 319, 330, 331, 332, 333, 334, 336, 338, 345, 346, 347, 348, 349, 350, 351, 353, 354, 355, 356, 357, 363, 364, 365, 402, 403, 404, 405, 406, 407, 414, 419, 420, 421, 469, 470, 471, 472, 474, 475, 476, 477, 479, 510, 511, 513, 514, 515, 516, 518, 560, 592, 593, 594, 595, 596, 597, 605, 606, 607, 609, 610, 611, 612, 613, 619, 621, 622, 624, 626, 627, 628, 630, 631, 633, 690, 702, 711, 712, 727, 731, 735, 744, 884, 885, 887, 901, 909, 910, 911, 912, 914, 916, 918, 919, 925, 926, 927, 947, 948, 952, 975, 1006, 1007, 1059, 1060, 1062, 1064, 1065, 1066, 1068, 1069, 1071, 1074, 1075, 1077, 1080, 1081, 1083, 1084, 1085, 1086, 1088, 1089, 1092, 1093, 1094, 1096, 1097, 1098, 1099, 1100, 1101, 1102, 1103, 1107, 1108, 1109, 1110, 1111, 1112, 1114, 1115, 1116, 1117, 1118, 1119, 1123, 1159, 1160, 1161, 1172, 1173, 1174, 1178, 1184, 1185, 1186, 1187, 1188, 1189, 1190, 1191, 1192, 1194, 1195, 1196, 1197, 1198, 1199, 1200, 1201, 1206, 1207, 1209, 1213, 1214, 1216, 1223, 1229, 1230, 1236, 1237, 1248, 1249, 1294, 1366, 1403, 1412, 1429, 1471, 1472, 1473, 1484, 1485, 1486, 1487, 1488, 1489, 1491, 1492, 1493, 1495, 1496, 1506, 1556, 1557, 1559, 1560, 1561, 1562, 1563, 1565, 1566, 1567, 1569, 1571, 1573, 1575, 1578, 1579, 1581, 1628, 1629, 1630, 1631, 1632, 1633, 1634, 1636, 1637, 1639, 1640, 1641, 1642, 1643, 1645, 1646, 1647, 1648, 1649, 1650, 1652, 1653, 1654, 1658, 1659, 1660, 1662, 1663, 1664, 1666, 1667, 1668, 1669, 1670, 1671, 1672, 1673, 1674, 1676, 1677, 1678, 1679, 1680, 1682, 1683, 1684, 1685, 1686, 1688, 1689, 1690, 1720, 1729, 1730, 1752, 1753, 1754, 1755, 1756, 1760, 1761, 1762, 1763, 1764, 1766, 1767, 1768, 1769, 1770, 1780, 1781, 1782, 1783, 1784, 1785, 1787, 1788, 1789, 1790, 1791, 1793, 1848, 1850, 1851, 1852, 1853, 1854, 1855, 1856, 1858, 1859, 1861, 1862, 1863, 1908, 1909]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPrepare.java (hunk 1 above)
Comment (mofeiatwork) [reply fe/fe-core/src/main/java/com/starrocks/qe/CoordinatorPrepare.java]: done

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 642 / 993 (64.65%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/CoordinatorPreprocessor.java|641|992|64.62%|[154, 158, 162, 170, 174, 178, 182, 190, 194, 202, 206, 214, 218, 222, 226, 230, 234, 238, 242, 250, 254, 258, 262, 270, 274, 305, 306, 307, 312, 313, 314, 315, 316, 317, 319, 330, 331, 332, 333, 334, 336, 338, 345, 346, 347, 348, 349, 350, 351, 353, 354, 355, 356, 357, 363, 364, 365, 402, 403, 404, 405, 406, 407, 414, 419, 420, 421, 469, 470, 471, 472, 474, 475, 476, 477, 479, 510, 511, 513, 514, 515, 516, 518, 560, 592, 593, 594, 595, 596, 597, 605, 606, 607, 609, 610, 611, 612, 613, 619, 621, 622, 624, 626, 627, 628, 630, 631, 633, 690, 702, 711, 712, 727, 731, 735, 744, 884, 885, 887, 901, 909, 910, 911, 912, 914, 916, 918, 919, 925, 926, 927, 947, 948, 952, 975, 1006, 1007, 1059, 1060, 1062, 1064, 1065, 1066, 1068, 1069, 1071, 1074, 1075, 1077, 1080, 1081, 1083, 1084, 1085, 1086, 1088, 1089, 1092, 1093, 1094, 1096, 1097, 1098, 1099, 1100, 1101, 1102, 1103, 1107, 1108, 1109, 1110, 1111, 1112, 1114, 1115, 1116, 1117, 1118, 1119, 1123, 1159, 1160, 1161, 1172, 1173, 1174, 1178, 1184, 1185, 1186, 1187, 1188, 1189, 1190, 1191, 1192, 1194, 1195, 1196, 1197, 1198, 1199, 1200, 1201, 1206, 1207, 1209, 1213, 1214, 1216, 1223, 1229, 1230, 1236, 1237, 1248, 1249, 1294, 1366, 1403, 1412, 1429, 1471, 1472, 1473, 1484, 1485, 1486, 1487, 1488, 1489, 1491, 1492, 1493, 1495, 1496, 1506, 1556, 1557, 1559, 1560, 1561, 1562, 1563, 1565, 1566, 1567, 1569, 1571, 1573, 1575, 1578, 1579, 1581, 1628, 1629, 1630, 1631, 1632, 1633, 1634, 1636, 1637, 1639, 1640, 1641, 1642, 1643, 1645, 1646, 1647, 1648, 1649, 1650, 1652, 1653, 1654, 1658, 1659, 1660, 1662, 1663, 1664, 1666, 1667, 1668, 1669, 1670, 1671, 1672, 1673, 1674, 1676, 1677, 1678, 1679, 1680, 1682, 1683, 1684, 1685, 1686, 1688, 1689, 1690, 1720, 1729, 1730, 1752, 1753, 1754, 1755, 1756, 1760, 1761, 1762, 1763, 1764, 1766, 1767, 1768, 1769, 1770, 1780, 1781, 1782, 1783, 1784, 1785, 1787, 1788, 1789, 1790, 1791, 1793, 1848, 1850, 1851, 1852, 1853, 1854, 1855, 1856, 1858, 1859, 1861, 1862, 1863, 1908, 1909]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**